server/src/main/java/org/elasticsearch/index/mapper/RuntimeFieldType.java

```diff
@@ -8,180 +8,53 @@
 
 package org.elasticsearch.index.mapper;
 
-import org.apache.lucene.analysis.TokenStream;
-import org.apache.lucene.search.MultiTermQuery;
-import org.apache.lucene.search.Query;
-import org.apache.lucene.search.spans.SpanMultiTermQueryWrapper;
-import org.apache.lucene.search.spans.SpanQuery;
-import org.elasticsearch.ElasticsearchException;
-import org.elasticsearch.common.CheckedBiConsumer;
-import org.elasticsearch.common.geo.ShapeRelation;
-import org.elasticsearch.common.time.DateMathParser;
-import org.elasticsearch.common.unit.Fuzziness;
 import org.elasticsearch.common.xcontent.ToXContentFragment;
 import org.elasticsearch.common.xcontent.XContentBuilder;
-import org.elasticsearch.index.query.SearchExecutionContext;
 
 import java.io.IOException;
-import java.time.ZoneId;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.Iterator;
 import java.util.List;
-import java.util.Locale;
 import java.util.Map;
 import java.util.function.BiFunction;
 
-import static org.elasticsearch.search.SearchService.ALLOW_EXPENSIVE_QUERIES;
-
 /**
- * Base implementation for a runtime field that can be defined as part of the runtime section of the index mappings
+ * Definition of a runtime field that can be defined as part of the runtime section of the index mappings
  */
-public abstract class RuntimeFieldType extends MappedFieldType implements ToXContentFragment {
-
-    private final CheckedBiConsumer<XContentBuilder, Boolean, IOException> toXContent;
-
-    protected RuntimeFieldType(String name, RuntimeFieldType.Builder builder) {
-        this(name, builder.meta(), builder::toXContent);
-    }
-
-    protected RuntimeFieldType(String name, Map<String, String> meta, CheckedBiConsumer<XContentBuilder, Boolean, IOException> toXContent) {
-        super(name, false, false, false, TextSearchInfo.SIMPLE_MATCH_WITHOUT_TERMS, meta);
-        this.toXContent = toXContent;
-    }
+public interface RuntimeFieldType extends ToXContentFragment {
```

Comment:
@romseygeek looks familiar? :) I did take inspiration from your previous proposal around aliases as runtime fields.

Reply:
I would propose renaming this to `RuntimeField` in a follow-up